### PR TITLE
RFR: Quiz Progress

### LINKED ISSUE #156: FE: Add Visual Progress to Quiz
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**
(taken from issue #122 )
Quiz should show a more guided approach -- a start screen, an ending screen, a progress bar.

**Additional context**
Add details, any other context or screenshots about the feature request here.

### PR DESCRIPTION
**Pull request by hungryflamingcat**
## Status: 

:rocket: Ready

## Description
1. make a progress bar to show quiz completion progress
2. make redo button next to progress bar
3. show final results as a progress  bar in redo modal
4. remove finish attempt and move functionality to "last" submit
    *last as in the last question actually answered, not the last question in the list

Related issues: closes #156 
Related PRs: #<number>

### CODE REVIEW

File: frontend/src/components/QuizRedo.js [hunk 1]
```diff
@@ -1,7 +1,35 @@
...
+  Modal,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  Progress
+} from 'reactstrap';
 
 class QuizRedo extends Component {
+  renderProgress = () => {
+    let correct = (this.props.numCorrectQ / this.props.numTotalQ) * 100;
```
Comment (Davidw1339) [inline frontend/src/components/QuizRedo.js]: nice work with the incorrect/correct :bar_chart: !!

File: frontend/src/components/QuizRedo.js [hunk 2]
```diff
@@ -10,8 +38,11 @@ class QuizRedo extends Component {
             You've completed the quiz!
           </ModalHeader>
           <ModalBody>
-            You've completed the quiz with a total score of:{' '}
-            {this.props.numCorrectQ}/{this.props.numTotalQ}.<br />
+            You've completed the quiz with a total score of:
```
Comment (Davidw1339) [inline frontend/src/components/QuizRedo.js]: nice modal! :computer:

File: frontend/src/components/QuizRedo.js [hunk 3]
```diff
@@ -1,7 +1,35 @@
...
 
 class QuizRedo extends Component {
+  renderProgress = () => {
+    let correct = (this.props.numCorrectQ / this.props.numTotalQ) * 100;
+    let incorrect =
+      ((this.props.numTotalQ - this.props.numCorrectQ) / this.props.numTotalQ) *
+      100;
+    return (
+      <div
+        style={{
```
Comment (Davidw1339) [inline frontend/src/components/QuizRedo.js]: not sure if necessary, but should this be a class or is it okay to be an inline style? @XuanLi-Leong

Comment (XuanLi-Leong) [reply frontend/src/components/QuizRedo.js]: Yeaaah if we could throw it into a class in quizpage.scss that'd be 💯

File: frontend/src/components/QuizViewer.js [hunk 4]
```diff
@@ -27,19 +34,62 @@ class QuizViewer extends Component {
...
+        <Progress color="success" value={value} />
+        {this.renderRedoButton()}
+      </div>
+    );
+  };
+
+  renderRedoButton = () => {
+    let color = '';
+    if (this.props.redoable === true) {
+      color = 'success';
```
Comment (XuanLi-Leong) [inline frontend/src/components/QuizViewer.js]: Do you mean color = 'primary' ?

File: frontend/src/components/QuizViewer.js [hunk 5]
```diff
@@ -78,7 +128,7 @@ class QuizViewer extends Component {
     );
   };
 
-  componentDidUpdate(prevProps) {
+  componentDidUpdate(prevProps, prevState) {
```
Comment (XuanLi-Leong) [inline frontend/src/components/QuizViewer.js]: can remove `prevState`

File: frontend/src/pages/QuizPage.js [hunk 6]
```diff
@@ -117,12 +117,14 @@ class QuizPage extends Component {
   };
 
   redoQuiz = questionProps => {
-    this.setState({
-      showEndModal: false,
-      redoable: false,
-      numCorrectQ: 0,
-      numRedo: this.state.numRedo + 1
-    });
+    if (this.state.redoable === true) {
```
Comment (XuanLi-Leong) [inline frontend/src/pages/QuizPage.js]: great check

**Pull request merged**